### PR TITLE
Subarray defringe

### PR DESCRIPTION
**Pull request by ariley1472**
I wrote up an example of how to defringe subarray data. I put it int he defringe_examples.rst, but let me know if it should go somewhere else. I also can't seem to get sphinx working, so I'm not entirely sure what it looks like whoops.

### CODE REVIEW

File: doc/source/defringe_examples.rst [hunk 1]
```diff
@@ -299,3 +299,111 @@ G750M Observation of AGK+81D266
 .. image:: output_14_0.png
 
 
+Subarray Data
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: ipython3
+    # Set up the data paths
```
Comment (dougbrn) [inline doc/source/defringe_examples.rst:306]: In testing this I found that there needs to be a spaces between the code:: ipython3 line and the following code in order to render the htmls correctly, the same isn't required for the parsed-literal blocks.

**Review by dougbrn — CHANGES_REQUESTED** [review]
This looks good! Just need to add spaces between the code:: lines and the example code in all relevant cases for the rendering.

**Review by dougbrn — APPROVED** [review]
With the formatting changes, this looks good!

**Comment by sean-lockwood** [comment]
I'm looking into the Python a bit.  Do you define `hdu` somewhere?:
`hdu.writeto(fullframe_output)`

File: doc/source/defringe_examples.rst [hunk 2]
```diff
@@ -299,3 +299,117 @@ G750M Observation of AGK+81D266
...
+    full_sci = np.zeros(np.shape(flat_data), dtype='float32')
+    full_sci = [int(ymin):int(ymax), :] = sci_data
+    # Use the same process to populate the error and data quality arrays.
+
+    header1 = fits.getheader(sci)
+    sci_header = fits.getheader(sci, 'SCI')
+    # Use the same process to populate the error and data quality headers
+
+    fullframe_output = f"{sci_file}_fullfield_crj.fits"
+    hdu.writeto(fullframe_output)
```
Comment (sean-lockwood) [inline doc/source/defringe_examples.rst:351]: `hdu` undefined.

**Comment by ariley1472** [comment]
@sean-lockwood It looks like I didn't include hdu. I'll make that change now

File: doc/source/defringe_examples.rst [hunk 3]
```diff
@@ -299,3 +299,117 @@ G750M Observation of AGK+81D266
...
+    flat_data = fits.getdata(f"{flat_file}_nsp.fits")
+
+    ymin_sci = sci[0].header['CENTERA2'] - (sci[0].header['SIZAXIS2']/2.)
+    ymax_sci = sci[0].header['CENTERA2'] + (sci[0].header['SIZAXIS2']/2.)
+
+    full_sci = np.zeros(np.shape(flat_data), dtype='float32')
+    full_sci = [int(ymin):int(ymax), :] = sci_data
+    # Use the same process to populate the error and data quality arrays.
+
+    header1 = fits.getheader(sci)
```
Comment (sean-lockwood) [inline doc/source/defringe_examples.rst:346]: I haven't seen getheader work on opened FITS files before.  Maybe this instead?:
`getheader(sci)` --> `sci[0].header`
`fits.getheader(sci, 'SCI')` --> `sci['SCI'].header`

File: doc/source/defringe_examples.rst [hunk 4]
```diff
@@ -299,3 +299,117 @@ G750M Observation of AGK+81D266
...
+                                  wavecal=f"{sci_file}_wav.fits")
+
+.. parsed-literal::
+    File written:  /Users/stisuser/data/path/odqf11060_crj.fits
+
+.. code:: ipython3
+
+    # If using G750L data, set the order sorter fringes in the flat field equal to one.
+    # This particular dataset is G750L data.
+    flat_hdu = fits.open(f"{flat_file}_nsp.fits", mode="update")
```
Comment (sean-lockwood) [inline doc/source/defringe_examples.rst:326]: `with fits.open(f"{flat_file}_nsp.fits", mode="update") as flat_hdu:`

File: doc/source/defringe_examples.rst [hunk 5]
```diff
@@ -299,3 +299,117 @@ G750M Observation of AGK+81D266
...
+    flat_data = flat_hdu[1].data
+    flat_data[:, :250] = 1
+    flat_hdu.flush()
```
Comment (sean-lockwood) [inline doc/source/defringe_examples.rst:329]: remove flush and close with `with` context above.

**Review by sean-lockwood — CHANGES_REQUESTED** [review]

**Comment by sean-lockwood** [comment]
Also, I'd pull the top comments out of the code blocks and make them plain-formatted text where appropriate.

**Comment by ariley1472** [comment]
@sean-lockwood I made all those changes.

**Review by sean-lockwood — APPROVED** [review]

**Pull request merged**